lib/plugin.js

```diff
@@ -214,15 +211,20 @@ ManifestPlugin.prototype.apply = function(compiler) {
       manifestMap[outputName].running = true;
       manifestMap[outputName].queue.shift()();
     }
-  }.bind(this);
+
+  };
 
   if (compiler.hooks) {
-    compiler.hooks.compilation.tap('ManifestPlugin', function (compilation) {
+    const SyncWaterfallHook = require('tapable').SyncWaterfallHook;
+
+    compiler.hooks.webpackManifestPluginAfterEmit = new SyncWaterfallHook(['manifest']);
+
+    compiler.hooks.compilation.tap('ManifestPlugin', compilation => {
       compilation.hooks.moduleAsset.tap('ManifestPlugin', moduleAsset);
     });
     compiler.hooks.emit.tap('ManifestPlugin', emit);
   } else {
-    compiler.plugin('compilation', function (compilation) {
+    compiler.plugin('compilation', compilation => {
```

Comment:
Shouldn't this also be `compiler.hooks.compilation.tap('ManifestPlugin', compilation)` ?

Reply:
Nah thats the legacy for webpack <=3

Reply:
Ah, ok.